### PR TITLE
【Comm】switch c_broadcast to broadcast

### PR DESCRIPTION
**Pull request by Xing-lil**
### PR Category

Communication Library

### PR Types

Improvements

### Description

Pcard-70448
additional code
https://github.com/PaddlePaddle/Paddle/pull/68863

### CODE REVIEW

File: python/paddle/distributed/auto_parallel/static/operators/dist_default.py [hunk 1]
```diff
@@ -80,16 +81,18 @@ def prim_operator_data_parallel_functor(ctx, src_op):
...
             attrs={
                 'ring_id': sync_group.id,
                 'root': 0,
-                'use_calc_stream': True,
                 OP_ROLE_KEY: OpRole.Forward,
             },
         )
+        new_op.dist_attr.execution_stream = (
+            AutoParallelStreamType.CALC_STREAM.value
+        )
```
Comment (liym27) [inline python/paddle/distributed/auto_parallel/static/operators/dist_default.py]: Please delete L93-95, don't set CALC_STREAM here.

File: python/paddle/distributed/auto_parallel/static/operators/dist_default.py [hunk 2]
```diff
@@ -610,16 +613,18 @@ def get_shape_attr_name():
...
                             attrs={
                                 'ring_id': sync_group.id,
                                 'root': 0,
-                                'use_calc_stream': True,
                                 OP_ROLE_KEY: OpRole.Forward,
                             },
                         )
+                        new_op.dist_attr.execution_stream = (
+                            AutoParallelStreamType.CALC_STREAM.value
+                        )
```
Comment (liym27) [inline python/paddle/distributed/auto_parallel/static/operators/dist_default.py]: Please delete L625-627, don't set CALC_STREAM here.

File: test/deprecated/auto_parallel/random_control_unittest_deprecated.py [hunk 3]
```diff
@@ -25,17 +25,9 @@
...
-from paddle import _legacy_C_ops
 from paddle.distributed.fleet import auto
 
 
-def dy_broadcast_helper(tensor):
-    _legacy_C_ops.c_broadcast(
-        tensor, tensor, 'root', 1, 'use_calc_stream', True, 'ring_id', 0
-    )
-    _legacy_C_ops.c_sync_calc_stream(tensor, tensor)
-
```
Comment (liym27) [inline test/deprecated/auto_parallel/random_control_unittest_deprecated.py:37]: why delete function "dy_broadcast_helper"

File: paddle/fluid/framework/new_executor/instruction/instruction_util.cc [hunk 4]
```diff
@@ -149,6 +149,8 @@ phi::DeviceContext* ParseDeviceContext(pir::Operation* op,
                 ->GetDevContext());
         dev_ctx->SetCommContext(comm_context);
         if (op_name.compare(paddle::dialect::ReduceScatterOp::name()) == 0 ||
+            op_name.compare(paddle::dialect::Broadcast_Op::name()) == 0 ||
```
Comment (liym27) [inline paddle/fluid/framework/new_executor/instruction/instruction_util.cc:152]: why `Broadcast_Op` ?

**Review by zyfncg — DISMISSED** [review]

**Review by zhangbo9674 — DISMISSED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Review by zhangbo9674 — APPROVED** [review]

**Pull request merged**